tutorials/Land/Soil/Coupled/equilibrium_test.jl

```diff
@@ -0,0 +1,556 @@
+aux = solver_config.dg.state_auxiliary;
+grads = solver_config.dg.state_gradient_flux
+ϑ_l = Q[:, ϑ_l_ind, :][:]
+z = aux[:, z_ind, :][:]
+T = aux[:, T_ind, :][:];
+
+
+# Gradients aren't calculated until the integration starts; we would
+# need to use the specified inital conditions to determine them. However,
+# we care more about showing that they tend to zero over time, as the system
+# approaches equlibrium, so we won't calculate them.
+
+# Save initial condition:
+initial_state = Dict{String, Array}(
+    "t" => [t],
+    "ϑ_l" => ϑ_l,
+    "T" => T,
+    "K∇h_vert" => [nothing],
+    "κ∇T_vert" => [nothing],
+);
+
+
+# We specify a function which evaluates `every_x_simulation_time` and returns
+# the state vector, appending the variables we are interested in into
+# `all_data`.
+
+# We also create an additional cartesian grid upon which an interpolated solution
+# of the DG output is evaluated. This is useful because the DG output is multi-valued
+# at element boundaries.
+zres = FT(0.02)
+boundaries = [
+    FT(0) FT(0) zmin
+    FT(1) FT(1) zmax
+]
+resolution = (FT(2), FT(2), zres)
+thegrid = solver_config.dg.grid
+intrp_brck = create_interpolation_grid(boundaries, resolution, thegrid)
+step = [1];
+callback = GenericCallbacks.EveryXSimulationTime(
+    every_x_simulation_time,
+) do (init = false)
+    t = ODESolvers.gettime(solver_config.solver)
+    iQ, iaux, igrads = interpolate_variables((Q, aux, grads), intrp_brck)
+    ϑ_l = iQ[:, ϑ_l_ind, :][:]
+    T = iaux[:, T_ind, :][:]
+    K∇h_vert = igrads[:, K∇h_vert_ind, :][:]
+    κ∇T_vert = igrads[:, κ∇T_vert_ind, :][:]
+    all_vars = Dict{String, Array}(
+        "t" => [t],
+        "ϑ_l" => ϑ_l,
+        "T" => T,
+        "K∇h_vert" => K∇h_vert,
+        "κ∇T_vert" => κ∇T_vert,
+    )
+    all_data[step[1]] = all_vars
+
+    step[1] += 1
+    nothing
+end;
+
+
+# # Run the integration
+ClimateMachine.invoke!(solver_config; user_callbacks = (callback,));
+
+
+# Get the final state and create plots:
+t = ODESolvers.gettime(solver_config.solver)
+iQ, iaux, igrads = interpolate_variables((Q, aux, grads), intrp_brck)
+
+ϑ_l = iQ[:, ϑ_l_ind, :][:]
+T = iaux[:, T_ind, :][:]
+K∇h_vert = igrads[:, K∇h_vert_ind, :][:]
+κ∇T_vert = igrads[:, κ∇T_vert_ind, :][:]
+all_vars = Dict{String, Array}(
+    "t" => [t],
+    "ϑ_l" => ϑ_l,
+    "T" => T,
+    "K∇h_vert" => K∇h_vert,
+    "κ∇T_vert" => κ∇T_vert,
+)
+all_data[n_outputs] = all_vars
+iz = iaux[:, z_ind, :][:]
+
+t = [all_data[k]["t"][1] for k in 1:n_outputs]
+t = ceil.(Int64, t ./ 60)
+
+ϑ_plot =
+    plot(initial_state["ϑ_l"], z, label = "t = 0", ylabel = "z", xlabel = "ϑ_l")
+plot!(all_data[1]["ϑ_l"], iz, label = "t = 0.75 days")
+plot!(all_data[2]["ϑ_l"], iz, label = "t = 1.5 days")
+plot!(all_data[3]["ϑ_l"], iz, label = "t = 2.25 days")
+plot!(all_data[4]["ϑ_l"], iz, label = "t = 3 days")
+
+
+K∇h_z_plot = plot(
+    all_data[1]["K∇h_vert"],
+    iz,
+    label = "0.75 days",
+    ytickfontcolor = "white",
+    xlabel = "K∇h_z (m/s)",
```

Comment:
It seems like we should be using `dict_of_nodal_states` and `export_state_plots`. If not, I think it would be worth generalizing this and moving into `src/Utilities/SingleStackUtils.jl` and or `docs/plothelpers.jl` (same with pieces in the callback).

Reply:
ok! Maybe we can chat briefly about this?

Reply:
ok! as per our discussion we will add into `dict_of_nodal_states` in a future branch. 
https://github.com/CliMA/ClimateMachine.jl/issues/1556